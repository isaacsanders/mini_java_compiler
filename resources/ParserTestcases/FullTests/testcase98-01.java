// Test 2
class Test2 {
	public static void main(String [] args) {
		boolean flag = true;
		int num1 = 1;
		int num2 = 5;

		while(num2 != num1) {
			num2 = num2 - 1;
		}

		if(flag) {
			System.out.println(num1);
		}
		else int num3 = 1;
	}
}
