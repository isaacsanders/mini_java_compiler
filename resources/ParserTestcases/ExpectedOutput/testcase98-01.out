StmtList ::= 
Type ::= BOOLEAN
LiteralExpr ::= TRUE
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Type ::= INT
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Type ::= INT
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr NE Expr
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr MINUS Expr
Stmt ::= ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
Stmt ::= WHILE LPAREN Expr RPAREN Stmt
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= SYSTEMOUTPRINTLN LPAREN Expr RPAREN SEMI
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
Type ::= INT
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
Stmt ::= IF LPAREN Expr RPAREN Stmt ELSE Stmt
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
Program ::= MainClassDecl ClassDeclList
